Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -159,6 +166,24 @@ private IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
         /// </summary>
         public IEnumerable<string> GetPipExecutionDirectoryOutputsEvents() => GetEventsByType(ExecutionEventId.PipExecutionDirectoryOutputs);
 
+        /// <summary>
+        /// Gets the total number of stored xlg events in the database
+        /// </summary>
+        public uint GetEventCount()
+        {
+            uint eventCount = 0;
+
+            Analysis.IgnoreResult(

```

Comment:
Can't you replace this whole method with
```csharp
    return Accessor.Use(database =>
	                {
	                    database.TryGetValue(Encoding.ASCII.GetBytes(EventCountKey), out var eventCountObj);
                        return EventCount.Parser.ParseFrom(eventCountObj).Value;
	                })
```

Reply:
The Accessor.Use method returns a possible int value, so I'm not sure if we can do a simplification like this.

Reply:
in that case:
```csharp
var maybeValue = Accessor.Use(...);
if (!maybeValue.Succeeded) maybeValue.Failure.Throw(); // or something like that
return maybeValue.Result;
```

Reply:
In other uses of the rocksDb store by say FingerprintStore.cs or the cache team seem to be doing it the same way as me (using the IgnoreResult, and having a var outside that is assigned to within the .Use( ... ) function. I think I should also follow this pattern then right?

Reply:
You are right that in other places this same pattern is used; for example, I found this in FingerprintStore.cs
```csharp
            var keyFound = false;
            Analysis.IgnoreResult(
                Accessor.Use(store =>
                {
                    keyFound = store.Contains(key, columnFamilyName);
                })
            );
            return keyFound;
```
That, however, doesn't necessarily make it right.  At face value, it looks ridiculous ignore explicitly returned result and instead from within a closure mutate some local var defined outside of the closure.  But there is more.  `Accessor.Use` can fail, in which case it returns a `Failure` object; when that happens, the failure is completely ignored here and instead the initial value assigned to `keyFound` (which is `false`) is returned.  Maybe the intention here is to exactly do that, i.e., if an error happens just ignore it and return the default value of `false`.  If so, this intention is completely obscured by this pattern; instead, if this code were written as
```csharp
var maybeFound = Accessor.Use(store => store.Contains(key, columnFamilyName));
return maybeFound.Succeeded 
    ? maybeFound.Result
    : false; // ignore the failure and return false
```
the intention would be 100% clear.

Code like this is pretty much the definition of spaghetti code.